sdk/src/Core/Amazon.Runtime/AmazonServiceClient.cs

```diff
@@ -159,11 +165,13 @@ protected AmazonServiceClient(AWSCredentials credentials, ClientConfig config)
                 _logger = Logger.GetLogger(this.GetType());
 
             config.Validate();
+            _config = config;
 
-            if(credentials != null)
-                config.DefaultAWSCredentials = credentials;
+            if (credentials != null)
```

Comment:
nit: Not sure why we need the `null` check. Can't you just always use `ExplicitAWSCredentials = credentials;` and if `credentials` is `null` then it just resets `ExplicitAWSCredentials` to null?